library/src/main/java/com/android/ex/chips/RecipientEditTextView.java

```diff
@@ -1259,7 +1263,14 @@ private boolean commitChip(int start, int end, Editable editable) {
                     QwertyKeyListener.markAsReplaced(editable, start, end, "");
                     CharSequence chipText = createChip(entry, false);
                     if (chipText != null && start > -1 && end > -1) {
-                        editable.replace(start, end, chipText);
+                        if (getRecipients().length < mMaxChips) {
+                            editable.replace(start, end, chipText);
+                        }else {
+                            editable.replace(start, end, "");
+                            if (mChipNotCreatedListener != null) {
+                                mChipNotCreatedListener.chipNotCreated();
```

Comment:
One improvement: lets pass in the `chipText` to our listener.

Just in case the implementer wanted to show a message of "chip not created for Luke Klinker" instead of just "chip not created".

Since we have the information readily available, there isn't much reason not to provide it.

Reply:
Do you think I should trim the chip text or let the user do that?

For example <John> will become John before being passed to the listener

Reply:
Good catch. Yes, let's trim it here, before passing it to the listener

Reply:
well looks like I can just pass text. I will do that and update.